cpp/oneapi/dal/backend/interop/table_conversion.hpp

```diff
@@ -114,6 +115,105 @@ inline daal::data_management::NumericTablePtr convert_to_daal_table(const table&
+    data.need_mutable_data();
+    column_indices.need_mutable_data();
+    row_indices.need_mutable_data();
+    const auto daal_data =
+        daal::services::SharedPtr<T>(data.get_mutable_data(), daal_object_owner{ data });
+    const auto daal_column_indices =
+        daal::services::SharedPtr<std::size_t>(column_indices.get_mutable_data(),
+                                               daal_object_owner{ column_indices });
+    const auto daal_row_indices =
+        daal::services::SharedPtr<std::size_t>(row_indices.get_mutable_data(),
+                                               daal_object_owner{ row_indices });
+
+    return daal::data_management::CSRNumericTable::create(
+        daal_data,
+        daal_column_indices,
+        daal_row_indices,
+        dal::detail::integral_cast<std::size_t>(column_count),
+        dal::detail::integral_cast<std::size_t>(row_count));
+}
+
+template <typename T>
+inline detail::csr_table convert_from_daal_csr_table(
+    const daal::data_management::CSRNumericTablePtr& nt) {
+    const std::int64_t row_count = nt->getNumberOfRows();
+    const std::int64_t column_count = nt->getNumberOfColumns();
+
+    T* daal_data{ nullptr };
+    std::size_t* daal_column_indices{ nullptr };
+    std::size_t* daal_row_indices{ nullptr };
+
+    nt->getArrays(&daal_data, &daal_column_indices, &daal_row_indices);
+
+    return detail::csr_table{ daal_data,
+                              daal_column_indices,
+                              daal_row_indices,
+                              row_count,
+                              column_count,
+                              detail::empty_delete<const T>(),
+                              detail::empty_delete<const std::size_t>(),
+                              detail::empty_delete<const std::size_t>() };
+}
+
+inline daal::data_management::CSRNumericTablePtr wrap_by_host_csr_adapter(
+    const detail::csr_table& table) {
+    const auto& dtype = table.get_metadata().get_data_type(0);
+
+    switch (dtype) {
+        case data_type::float32: return host_csr_table_adapter<float>::create(table);
+        case data_type::float64: return host_csr_table_adapter<double>::create(table);
+        case data_type::int32: return host_csr_table_adapter<std::int32_t>::create(table);
+        default: return daal::data_management::CSRNumericTablePtr();
+    }
+}
+
+template <typename Float>
+inline daal::data_management::CSRNumericTablePtr copy_to_daal_csr_table(
+    const detail::csr_table& table) {
+    auto block = detail::csr_accessor<const Float>{ table }.pull();
+    return convert_to_daal_csr_table(block.data,
+                                     block.column_indices,
+                                     block.row_indices,
+                                     table.get_row_count(),
+                                     table.get_column_count());
+}
+
+template <typename Float>
+inline daal::data_management::CSRNumericTablePtr convert_to_daal_table(
```

Comment:
this logic shall be enabled via general `convert_to_daal_table(const table&)` function